src/pymor/parameters/functionals.py

```diff
@@ -325,3 +326,134 @@ def evaluate(self, mu=None):
+        assert np.all(thetas_mu_bar > 0)
+        assert isinstance(alpha_mu_bar, Number)
+        assert alpha_mu_bar > 0
+        self.__auto_init(locals())
+        self.thetas_mu_bar = thetas_mu_bar # why is this required after __auto_init?
+
+    def evaluate(self, mu=None):
+        mu = self.parse_parameter(mu)
+        thetas_mu = np.array([theta(mu) for theta in self.thetas])
+        assert np.all(thetas_mu > 0)
+        return self.alpha_mu_bar * np.min(thetas_mu / self.thetas_mu_bar)
+
+    def d_mu(self, component, index=()):
+        raise NotImplementedError
+
+
+class MaxThetaParameterFunctional(ParameterFunctional):
+    """|ParameterFunctional| implementing the max-theta approach from [Haa17]_ (Exercise 5.12).
+
+    Let V denote a Hilbert space and let a: V x V -> K denote a continuous bilinear form or l: V -> K a continuous
+    linear functional, either with affine
+    decomposition
+
+      a(u, v, mu) = sum_{q = 1}^Q theta_q(mu) a_q(u, v)  or  l(v, mu) = sum_{q = 1}^Q theta_q(mu) l_q(v)
+
+    for Q coefficient |ParameterFunctional|s theta_1, ..., theta_Q and continuous bilinear forms
+    a_1, ..., a_Q: V x V -> K or continuous linear functionals l_q: V -> K. Let mu_bar be a parameter with respect to
+    which the continuity constant of a(., ., mu_bar) or l(., mu_bar)is known, i.e. we known gamma_mu_bar > 0, s.t.
+
+      a(u, v, mu_bar) <= gamma_mu_bar |u|_V |v|_V  or  l(v, mu_bar) <= gamma_mu_bar |v|_V.
+
+    The max-theta approach from [Haa17]_ (Exercise 5.12) allows to obtain a computable bound for the continuity
+    constant of a(., ., mu) or l(., mu) for arbitrary parameters mu, since
+
+      a(u, v, mu=mu) <= |max_{q = 1}^Q theta_q(mu)/theta_q(mu_bar)|  |a(u, v, mu=mu_bar)|
+
+    or
+
+      l(v, mu=mu) <= |max_{q = 1}^Q theta_q(mu)/theta_q(mu_bar)| |l(v, mu=mu_bar)|,
+
+    if all theta_q(mu_bar) != 0.
+
+    Given a list of the thetas, the parameter mu_bar and the constant gamma_mu_bar, this functional thus evaluates
+    to ::
+
+      gamma_mu_bar * max{q = 1}^Q theta_q(mu)/theta_q(mu_bar)
+
+
+    Parameters
+    ----------
+    thetas
+        List or tuple of |ParameterFunctional|s
+    mu_bar
+        Parameter associated with gamma_mu_bar.
+    gamma_mu_bar
+        Known continuity constant.
+    name
+        Name of the functional.
+    """
+
+    def __init__(self, thetas, mu_bar, gamma_mu_bar=1., name=None):
+        assert isinstance(thetas, (list, tuple))
+        assert len(thetas) > 0
+        assert all([isinstance(theta, (Number, ParameterFunctional)) for theta in thetas])
+        thetas = [ConstantParameterFunctional(f) if not isinstance(f, ParameterFunctional) else f
+                  for f in thetas]
+        self.build_parameter_type(*[t for t in thetas if t.parametric])
+        mu_bar = self.parse_parameter(mu_bar)
+        thetas_mu_bar = np.array([theta(mu_bar) for theta in thetas])
+        assert np.all(np.logical_or(thetas_mu_bar < 0, thetas_mu_bar > 0))
```

Comment:
`thetas_mu_bar != 0` ?

Reply:
I used this due to float cmp problems, using float_cmp now instead.